### ISSUE TITLE
Refactor the Workstation driver check

### ISSUE THREAD

**Issue by tenthirtyam** [hashicorp/packer-plugin-vmware#242]
#### Overview of the Issue

When using VMware Workstation 17.5.2 and the personal license the drive can be mistake as Player.

When using the personal license, there is no license file in `/etc/vmware/license-ws*`, thus it skips the Workstation drivers and uses Player; thereby updating the driver state and app path from `/usr/bin/vmware` to `/usr/bin/player`. Then it will try to use the `/usr/bin/vmrun -T player list` instead of `/usr/bin/vmrun -T ws lis`t and that is where it fails.

To workaround the issue, you can run the following:

```console
sudo touch /etc/vmware/license-ws-foo
```

However, the following will need to be addressed:

- Alternative method to confirm us of Workstation vs Player.
- Determine if the license check is still needed.

### Plugin and Packer versions

- `hashicorp/packer` v1.11.2 and earlier.
- `hashicorp/packer-pluign-vmware` v1.1.0 and earlier.

**Comment by tenthirtyam** [comment]
[**VMware Fusion and Workstation are Now Free for All Users**](https://blogs.vmware.com/cloud-foundation/2024/11/11/vmware-fusion-and-workstation-are-now-free-for-all-users/)

**TL;DR:**

> _"We’re thrilled to announce a significant change that reflects our commitment to making VMware Fusion and VMware Workstation more accessible than ever. Starting November 11, 2024, these powerful desktop hypervisor products will be available for free to everyone—commercial, educational, and personal users alike."_

Based upon this change, for license check will be removed from the plugin for the latest versions for Fusion and Workstation.

**Comment by tenthirtyam** [comment]
This is likely because this version removes the license as both Workstation and Fusiom are now free. 

I'll be addressing this in a PR that is still a WIP but delayed on my part. Rolling back to an earlier version should hopefully workaround the issue.

**Comment by tenthirtyam** [comment]
Submitted [PR 277](https://github.com/hashicorp/packer-plugin-vmware/pull/277) for review.
